gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapper.java

```diff
@@ -0,0 +1,216 @@
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ReferralRequest;
+import org.hl7.fhir.dstu3.model.ReferralRequest.ReferralRequestStatus;
+import org.hl7.fhir.dstu3.model.ReferralRequest.ReferralCategory;
+import org.hl7.fhir.dstu3.model.UriType;
+import org.hl7.v3.CD;
+import org.hl7.v3.CV;
+import org.hl7.v3.II;
+import org.hl7.v3.IVLTS;
+import org.hl7.v3.RCMRMT030101UK04AgentRef;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04Participant;
+import org.hl7.v3.RCMRMT030101UK04Participant2;
+import org.hl7.v3.RCMRMT030101UK04RequestStatement;
+import org.hl7.v3.RCMRMT030101UK04ResponsibleParty3;
+import org.hl7.v3.TS;
+
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+
+public class ReferralRequestMapper {
+    private static final String META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-ReferralRequest-1";
+    private static final String IDENTIFIER_SYSTEM = "https://PSSAdaptor/";
+    private static final String PRIORITY_PREFIX = "Priority: ";
+    private static final String ACTION_DATE_PREFIX = "Action Date: ";
+    private static final String PRACTITIONER_REFERENCE_PREFIX = "Practitioner/";
+    private static final String RESP_PARTY_TYPE_CODE = "RESP";
+    private static final String PRF_TYPE_CODE = "PRF";
+    private static final String PPRF_TYPE_CODE = "PPRF";
+
+    public ReferralRequest mapToReferralRequest(RCMRMT030101UK04EhrComposition ehrComposition,
+        RCMRMT030101UK04RequestStatement requestStatement) {
+        var id = requestStatement.getId().get(0).getRoot();
+        var identifier = getIdentifier(id);
+        var notes = getNotes(requestStatement);
+        var reasonCode = getReasonCode(requestStatement.getCode());
+        var authoredOn = getAuthoredOn(requestStatement.getAvailabilityTime());
+        var recipient = getRecipient(requestStatement.getResponsibleParty());
+        var requester = getRequesterAgentReference(requestStatement.getParticipant(), ehrComposition);
+
+        /**
+         * TODO: Known future implementations to this mapper
+         * - subject: references a global patient resource for the transaction (NIAD-2024)
+         * - context: references an encounter resource if it has been generated from the ehrComposition (NIAD-2025)
+         * - concatenate source practice org id to identifier URL (NIAD-2021)
+         */
+
+        return createRequestStatement(id, identifier, notes, reasonCode, authoredOn, recipient, requester);
+    }
+
+    private Identifier getIdentifier(String id) {
+        Identifier identifier = new Identifier()
+            .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL (NIAD-2021)
+            .setValue(id);
+        return identifier;
+    }
+
+    private CodeableConcept getReasonCode(CD reasonCode) {
+        if (reasonCode != null) {
+            return new CodeableConceptMapper().mapToCodeableConcept(reasonCode);
```

Comment:
I'm sorry Barry, I've asked Elliot for the same thing in his PR, but I'm not sure which one will be merged first, so I'll add this here as well :D If you could annotate CodeableConceptMapper class with @Service annotation, and inject it in your mapper, instead of doing`new CodeableConceptMapper()`? I forgot to add this annotation to CodeableConceptMapper earlier :( And then you could mock it in ReferralRequestMapperTest